src/app/qgsmaptoolselectionhandler.cpp

```diff
@@ -101,12 +101,15 @@ QgsMapToolSelectionHandler::QgsMapToolSelectionHandler( QgsMapCanvas *canvas, Qg
   : mCanvas( canvas )
   , mSelectionMode( selectionMode )
   , mSnapIndicator( qgis::make_unique< QgsSnapIndicator >( canvas ) )
+  , mIdentifyMenu( new QgsIdentifyMenu( mCanvas ) )
```

Comment:
Should this be a `std::unique_ptr`?

Reply:
Nope -- have discussed with @PeterPetrik , but the constructor of QgsIdentifyMenu blocks this.

Reply:
leaving ownership to canvas (as in the comment below) sounds good. but how does the constructor block this?

Reply:
```
QgsIdentifyMenu::QgsIdentifyMenu( QgsMapCanvas *canvas )
  : QMenu( canvas )
  , mCanvas( canvas )
...
```

Reply:
so?

Reply:
you are not able to specify different (e.g. `nullptr`) parent to `QMenu` and `mCanvas` member, so if you delete `mIdentify` in `QgsMapToolSelectionHandler` it is kind of double-delete, no?

Reply:
I think that the menu SHOULD be parented to the handler. That's the real parent and owner of the menu. But the constructor of QgsIdentifyMenu doesn't accept a non-map canvas parent.

Reply:
In this case it's about clear ownership management and not the unique_ptr itself if I understand this correctly? Makes sense.

Reply:
Would the following constructor improve the situation?

```
QgsIdentifyMenu::QgsIdentifyMenu( QgsMapCanvas *canvas, QObject *parent = nullptr )
  : QMenu( parent ? parent : canvas )
  , mCanvas( canvas )
...
```